Tests/Enable-BMEnvironment.Tests.ps1

```diff
@@ -0,0 +1,118 @@
+
+#Requires -Version 5.1
+Set-StrictMode -Version 'Latest'
+
+& (Join-Path -Path $PSScriptRoot -ChildPath 'Initialize-Tests.ps1' -Resolve)
+
+$session = New-BMTestSession
+
+function Init
+{
+    $Global:Error.Clear()
+    Get-BMEnvironment -Session $session | Disable-BMEnvironment -Session $session
+}
+
+function GivenEnvironment
+{
+    param(
+        [Parameter(Mandatory)]
+        [string]$Named,
+
+        [Switch]$Disabled
+    )
+
+    New-BMEnvironment -Session $session -Name $Named -ErrorAction Ignore
+    if( $Disabled )
+    {
+        Disable-BMEnvironment -Session $session -Name $Named
+    }
+    else
+    {
+        Enable-BMEnvironment -Session $session -Name $Named
+    }
+}
+
+function ThenEnvironmentEnabled
+{
+    param(
+        [Parameter(Mandatory)]
+        [string]$Named
+    )
+
+    $environment = Get-BMEnvironment -Session $session -Name $Named 
+    $environment | Should -Not -BeNullOrEmpty
+    $environment.active | Should -BeTrue
+}
+
+function ThenEnvironmentDisabled
+{
+    param(
+        [Parameter(Mandatory)]
+        [string]$Named
+    )
+
+    $environment = Get-BMEnvironment -Session $session -Name $Named 
+    $environment | Should -Not -BeNullOrEmpty
+    $environment.active | Should -BeFalse
+}
+
+function WhenEnablingEnvironment
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory)]
+        [string]$Named,
+
+        [Switch]
+        $WhatIf
+    )
+
+    $optionalParams = @{ }
+    if( $WhatIf )
+    {
+        $optionalParams['WhatIf'] = $true
+    }
+
+    $result = Enable-BMEnvironment -Session $session -Name $Named @optionalParams
+    $result | Should -BeNullOrEmpty
+}
+
+Describe 'Enable-BMEnvironment.when environment is enabled' {
+    It ('should enable the environment') {
+        Init
+        GivenEnvironment -Named 'Fubar' -Disabled

```

Comment:
Describe is `Enable-BMEnvironment.when environment is enabled` but environment starts off disabled.